### ISSUE TITLE
Layout issues in Chrome Canary

### ISSUE THREAD

**Issue by justindonnelly** [DestinyItemManager/DIM#2915]
I've been seeing issues with the page layout in Chrome Canary for probably about a month (apologies for not reporting this sooner). I see the same issues on both Windows and OSX and on both app.destinyitemmanager.com and beta.destinyitemmanager.com.

The root issue looks like the header and footer are not remaining pinned to the top and the bottom of the window. So when I scroll down in the page, the links and controls and search box are no longer visible. However, the character emblems do remain in the correct position (see screenshot below).

Similarly, the item comparison UI and loadout UI are at the bottom of the page, not pinned to the bottom of the window. So when you open them, it appears as if nothing happened because they're not visible unless you happen to have scrolled all the way to the bottom of the page.

<img width="1061" alt="screen shot 2018-06-30 at 2 06 54 pm" src="https://user-images.githubusercontent.com/18452875/42129068-e4ece122-7c6e-11e8-8202-d54157d5f053.png">

**Comment by justindonnelly** [comment]
Never mind. I cleared cookies and it's working correctly now.

**Comment by justindonnelly** [comment]
As soon as I went to update my settings after clearing cookies, it started happening again.

I turns out that this reliably happens when I have the color blindness setting enabled. It doesn't seem to matter which setting I use, anything other than the default produces the issue described above.

**Comment by duckmanBR** [comment]
Errors on Canary probably are Canary fault.  Tested on Chrome Beta and it's working.

**Comment by justindonnelly** [comment]
Yeah, I'm seeing the issue in Beta also (68.0.3440.42 (Official Build) beta (64-bit)). I should have double checked before reporting it as Canary-only. I think it started happening just before version 68 was promoted to Beta on about June 07.

**Comment by justindonnelly** [comment]
Note that 68 is scheduled to go to Stable on July 24.

**Comment by duckmanBR** [comment]
Really sorry.  I'm so used on being on Chrome Beta on mobile that completely forgot that I'm on stable on desktop.
Reproduced on Chrome Beta Android (68.0.3440.40) on both modes, mobile and "desktop site".

**Comment by bhollis** [comment]
Does that repro on Chrome Beta Desktop? Trying to figure out how much stuff I’ll need to install to figure this out.

**Comment by GriceTurrble** [comment]
Bump for visibility, coming here with this issue now on the Stable Desktop release of Chrome (68.0.3440.106).

Problem's root appears to be the `filter` CSS setting on the `<body>` tag, which is being set with the color blindness option from settings (via a variable `--color-filter` set in the style attribute of `html` tag).

Toggling that CSS creates the issue:

![dim_filter_effect](https://user-images.githubusercontent.com/9964605/43974585-08aea2a8-9ca9-11e8-98e8-4a1170b4cf45.gif)

I can't seem to freeze the screen when it's loading, but the issue exists when the color filter is applied, so that appears related.

---

This may be in relation to a similar bug in FireFox, discussed on this SO question: https://stackoverflow.com/questions/42049211/css-filter-breaking-positioning-in-firefox

Because the filter is being applied to the body tag, it is altering which tag is used as the parent target for `position: sticky` containers. Thus, the div `#header` is not sticking where it's supposed to.

Not exactly sure how to correct for this, as it's a rendering bug in the browsers. Getting around it might require wrapping the content sections in divs that get the color filters, as opposed to applying filters to the body.

**Comment by bhollis** [comment]
Note that in the redux branch I’m changing a lot of how this header works, so I’ll reassess this bug after that merges.

**Comment by Mulchman** [comment]
Thanks @GriceTurrble - turning colorblind stuff off fixes this temporarily. I look forward to a proper solution.

**Comment by GriceTurrble** [comment]
Checked out 4.70.1 on Chrome 68.0.3440.106, works fine now, thank you. 👍
